enhancements/installer/component-selection.md

```diff
@@ -0,0 +1,358 @@
+we can add their component names to a list of default-disabled components that the installer populates
+the install-config with.  Users can then edit that disabled list in the install-config to remove those
+components if they want them enabled, and add other components they'd like to disable.
+
+In the future, we might allow specific APIs to be disabled, such as the build api.  This could be done by
+defining additional capability keywords that can be put into the install-config field being defined here,
+which would drive the creation of cluster config that the apiserver+controllers used to disable that particular
+api.
+
+
+### Risks and Mitigations
+
+The primary risk is that teams understand how to use these new annotations and apply them
+correctly to the full set of resources that make up their component.  Inconsistent or
+partial labeling will result in inconsistent or partially deployed resources for a component.
+
+Another risk is that this introduces more deployment configurations which might
+have unforeseen consequences (e.g. not installing the imageregistry causes some
+other component that assumes there is always an imageregistry or assumes the
+presence of some CRD api that is installed with the imageregistry to break).
+
+There was some discussion about the pros/cons of allowing each component to be enabled/disabled independent
+of that component explicitly opting into a particular (presumably well tested) configuration/topology
+[here](https://github.com/openshift/enhancements/pull/200#discussion_r375837903).  The position of this EP is that
+we should only recommend the exclusion of fully independent "capability" components that are not depended on by
+other components.  Further the assumption is that it will be reasonable to tell a customer who disabled
+something and ended up with a non-functional cluster that their chosen exclusions are simply not supported
+currently, or that they must accept the degraded functionality caused by the missing dependency, if they
+intend to keep it disabled.
+
+Since the only components/resources that can be filtered out of the installation are ones that are explicilty
+annotated with `capability.openshift.io/$component`, end-users will not be able to use this mechanism to filter
+components/resources that we did not intend for them to be able to filter out.
+
+## Design Details
+
+### Open Questions
+
+
+1. Do we want to constrain this functionality to turning off individual components?  We could
+also use it to
+  a) turn on/off groups of components as defined by "solutions" (e.g. a "headless" solution
+  which might turn off the console but also some other components).  This is what CLUSTER_PROFILES
+  sort of enable, but there seems to be reluctance to expand the cluster profile use case to include
+  these sorts of things.
+  b) enable/disable specific configurations such as "HA", where components could contribute multiple
```

Comment:
I feel we already have this with the two availability policies on Infra CR…

Reply:
only listed here for completeness, not really considering using this functionality to achieve HA vs non-HA configuration of components at this point